script.js

```diff
@@ -1,4 +1,341 @@
+// Deck is shuffled.
+// User clicks Submit to deal cards.
+// The cards are analysed for game winning conditions, e.g. Blackjack.
+// The cards are displayed to the user.
+// The user decides whether to hit or stand, using the submit button to submit their choice.
+// The user's cards are analysed for winning or losing conditions.
+// The computer decides to hit or stand automatically based on game rules.
+// The game either ends or continues.
+
+// first create a global variables
+var cardDeck = [];
+var shuffledDeck = [];
+var playerHand = [];
+var dealerHand = [];
+var winner = ``;
+
+var player = "player";
+var dealer = "dealer";
+var hit = "hit";
+var stand = "stand";
+var playersTotalPoints = 0;
+var dealersTotalPoints = 0;
+
+var gameMode = "starting";
+var starting = "starting";
+var dealCard = "dealCard";
+var playerTurn = "playerTurn";
+var dealerTurn = "dealerTurn";
+var checkWinner = "checkWinner";
+
+var generateDeck = function () {
+  // create an array of 4 suits in the deck
+  var suits = ["Diamonds", "Clubs", "Hearts", "Spades"];
+
+  // while function to loop over the suits array
+  var suitIndex = 0;
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+
+    // loop from 1 to 13 to create all the cards for the suits
+
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      // by default card name is the same as rankcounter
+      var cardName = rankCounter;
+      var cardPoint = rankCounter;
+
+      // set the card value for ace, king, wueens
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+        cardPoint = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        cardPoint = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        cardPoint = 10;
+      }
+      // create a new card with the current name, suit and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        point: cardPoint,
+      };
+      // add new card to the deck
+      cardDeck.push(card);
```

Comment:
You should not be adding a card to the global variable cardDeck. Instead, you should have a local variable inside this function and push a card to the local variable. 
Then, at the end of the function, return this local variable. Always avoid manipulating global variables wherever possible. Also, if you chose to update a global variable, then your return statement in this function is not necessary as well. Looking at line 316, you did not use the return value of this function.